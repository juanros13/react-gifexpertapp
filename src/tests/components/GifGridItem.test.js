import React from 'react';
import { shallow } from 'enzyme'

import  GifGridItem  from '../../components/GifGridItem';


describe('Pruebas en el <GifGridItem />', () => {
    
    const titulo = 'Esto es un titulo';
    const url = 'http://localhost/algo.jpg';
    let wrapper = shallow( <GifGridItem title={ titulo } url = { url } /> );
    
    // beforeEach( () =>  {

    //      wrapper = shallow( <GifGridItem /> );

    // });



    test('debe de mostrar <GifGridItem /> correctamente', () => {
       

        
        expect( wrapper ).toMatchSnapshot();

    });
    test('Debe tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(titulo);
    });
    test('Debe de tener la imagen al url y alr de los props', () => {
        const img = wrapper.find('img');
        console.log(img.prop('src'));
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( titulo );
    });
    test('El div debe tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        
        // console.log(div.prop('className').split(' ')[2]);
        // expect(div.prop('className').split(' ')[2]).toBe('animate__fadeIn');\
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true); 


        //expect(div.prop('class'))
        
    })
    
    
})
