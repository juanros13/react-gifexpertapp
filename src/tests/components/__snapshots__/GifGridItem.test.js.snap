// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGridItem /> debe de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Esto es un titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
    Esto es un titulo
  </p>
</div>
`;
